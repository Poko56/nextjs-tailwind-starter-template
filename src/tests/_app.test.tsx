import { render, fireEvent } from '@testing-library/react'

import '@tests/test-utils'
import Home from '@pages/index'

describe('Home page', () => {
  it('should render Home Page', () => {
    render(<Home />)
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('clicking button triggers alert', () => {
    const { getByText } = render(<Home />, {})
    window.alert = jest.fn()
    fireEvent.click(getByText('Test Button'))
    expect(window.alert).toHaveBeenCalledWith('Button is Clicked!')
  })
})
