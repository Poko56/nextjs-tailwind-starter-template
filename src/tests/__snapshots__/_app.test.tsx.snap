// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="bg-white border-b border-gray-200"
    >
      <div
        class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
      >
        <div
          class="flex justify-between h-16"
        >
          <div
            class="flex"
          >
            <div
              class="flex-shrink-0 flex items-center"
            >
              <h1
                class="text-xl font-bold text-gray-700 hover:text-gray-800"
              >
                <a
                  href="/"
                >
                  Demo Site
                </a>
              </h1>
            </div>
            <div
              class="hidden sm:-my-px sm:ml-12 sm:flex sm:space-x-8"
            >
              <a
                class="inline-flex items-center px-1 pt-1 border-b-2 text-sm font-medium border-indigo-500 text-gray-900"
                href="/"
              >
                Home
              </a>
              <a
                class="inline-flex items-center px-1 pt-1 border-b-2 text-sm font-medium border-transparent text-gray-500 hover:border-gray-300 hover:text-gray-700"
                href="/about"
              >
                About
              </a>
            </div>
          </div>
          <div
            class="hidden sm:ml-6 sm:flex sm:items-center"
          >
            <button
              class="bg-white p-1 rounded-full text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
            >
              <span
                class="sr-only"
              >
                View notifications
              </span>
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 17h5l-1.405-1.405A2.032 2.032 0 0118 14.158V11a6.002 6.002 0 00-4-5.659V5a2 2 0 10-4 0v.341C7.67 6.165 6 8.388 6 11v3.159c0 .538-.214 1.055-.595 1.436L4 17h5m6 0v1a3 3 0 11-6 0v-1m6 0H9"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
            <div
              class="ml-3 relative"
            >
              <div>
                <button
                  aria-haspopup="true"
                  class="max-w-xs bg-white flex items-center text-sm rounded-full focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
                  id="user-menu"
                >
                  <span
                    class="sr-only"
                  >
                    Open user menu
                  </span>
                  <img
                    alt=""
                    class="h-8 w-8 rounded-full"
                    src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="-mr-2 flex items-center sm:hidden"
          >
            <button
              class="bg-white inline-flex items-center justify-center p-2 rounded-md text-gray-400 hover:text-gray-500 hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
            >
              <span
                class="sr-only"
              >
                Open main menu
              </span>
              <svg
                aria-hidden="true"
                class="block h-6 w-6"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 6h16M4 12h16M4 18h16"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="hidden h-6 w-6"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6 18L18 6M6 6l12 12"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="hidden sm:hidden"
      >
        <div
          class="pt-2 pb-3 space-y-1"
        >
          <a
            class="border-transparent text-gray-600 hover:bg-gray-50 hover:border-gray-300 hover:text-gray-800 block pl-3 pr-4 py-2 border-l-4 text-base font-medium"
            href="/"
          >
            Home
          </a>
          <a
            class="border-transparent text-gray-600 hover:bg-gray-50 hover:border-gray-300 hover:text-gray-800 block pl-3 pr-4 py-2 border-l-4 text-base font-medium"
            href="/about"
          >
            About
          </a>
        </div>
        <div
          class="pt-4 pb-3 border-t border-gray-200"
        >
          <div
            class="flex items-center px-4"
          >
            <div
              class="flex-shrink-0"
            >
              <img
                alt=""
                class="h-10 w-10 rounded-full"
                src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
              />
            </div>
            <div
              class="ml-3"
            >
              <div
                class="text-base font-medium text-gray-800"
              >
                Tom Cook
              </div>
              <div
                class="text-sm font-medium text-gray-500"
              >
                tom@example.com
              </div>
            </div>
            <button
              class="ml-auto bg-white flex-shrink-0 p-1 rounded-full text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
            >
              <span
                class="sr-only"
              >
                View notifications
              </span>
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                x-description="Heroicon name: bell"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 17h5l-1.405-1.405A2.032 2.032 0 0118 14.158V11a6.002 6.002 0 00-4-5.659V5a2 2 0 10-4 0v.341C7.67 6.165 6 8.388 6 11v3.159c0 .538-.214 1.055-.595 1.436L4 17h5m6 0v1a3 3 0 11-6 0v-1m6 0H9"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
          <div
            class="mt-3 space-y-1"
          >
            <a
              class="block px-4 py-2 text-base font-medium text-gray-500 hover:text-gray-800 hover:bg-gray-100"
              href="/"
              role="menuitem"
            >
              Your Profile
            </a>
            <a
              class="block px-4 py-2 text-base font-medium text-gray-500 hover:text-gray-800 hover:bg-gray-100"
              href="/"
              role="menuitem"
            >
              Settings
            </a>
            <a
              class="block px-4 py-2 text-base font-medium text-gray-500 hover:text-gray-800 hover:bg-gray-100"
              href="/"
              role="menuitem"
            >
              Sign Out
            </a>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="container"
    >
      <main
        class="main"
      >
        <h1
          class="title"
        >
          Home Page
        </h1>
        <div
          class="pt-10"
        >
          <button
            class="inline-flex items-center px-4 py-2 border border-transparent text-base font-medium rounded-md shadow-sm text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
            type="button"
          >
            Test Button
          </button>
        </div>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
